packages/ra-core/src/controller/edit/useEditController.ts

```diff
@@ -151,7 +151,7 @@ export const useEditController = <
                 : '',
     });
 
-    const recordCached = { id, previousData: record };
+    const recordCached = { id, previousData: record, meta: mutationMeta };
```

Comment:
This fixes the issue when the `meta` prop is passed during hook declaration time (i.e. when passed as `mutationOptions` in `<Edit>`), but won't work if the `meta` prop is passed at calltime (i.e. passed to the `<SaveButton>` for instance). I think we should instead pass the calltime `meta` to `updateCache()`, which would work in all cases.